model/driver/src/icon4py/model/driver/icon4py_driver.py

```diff
@@ -226,6 +226,11 @@ def _do_dyn_substepping(
                 f"simulation date : {self._simulation_date} substep / n_substeps : {dyn_substep} / "
                 f"{self.n_substeps_var} , is_first_step_in_simulation : {self._is_first_step_in_simulation}"
             )
+            self.solve_nonhydro.update_time_levels_for_velocity_tendencies(
```

Comment:
Can you move that function out  `solve_nonhydro`? As the swapping is no longer in the dycore this seems to be the wrong place. It is part of the driving code. So rather belongs here and not anymore in the dycore.

Reply:
Should we extract the `do_dyn_substep` from the driver? At least in tests we duplicate the swapping logic when we test the full timestep? Maybe in a later cleanup?